README.md

```diff
@@ -205,45 +293,87 @@ This method returns a promise that resolves when all the promises in the chain r
 
 The parameter *series* is an array of promises and/or functions that return promises.
 
-For example, in the following code *p* rejects with value `"2"` in 2.5 seconds:
+The action function is triggered at the moment when the Promise instance is created. So using functions returning Promise instances to pass into `Promise.serial` makes instantiation sequential. I.e. a promise is created and the action is triggered only when the previous Promise in the series got resolved or rejected.
+
+For example, in the following code `p` rejects with value `"2"` in 2.5 seconds
+(the second function-argument is executed only when the first Promise resolves and the second one is instantiated):
 
 ```squirrel
 local series = [
-    Promise(@(resolve, reject) imp.wakeup(1, @() resolve(1))),
-    @() Promise(@(resolve, reject) imp.wakeup(1.5, @() reject(2))),
-    Promise(@(resolve, reject) imp.wakeup(0.5, @() resolve(3)))
+    Promise(
+        function(resolve, reject) {
+            imp.wakeup(1,
+                function() {
+                    resolve(1)
+                }
+            )
+        }
+    ),
+    function() {
+        return Promise(
+            function(resolve, reject) {
+                imp.wakeup(1.5,
+                    function() {
+                        resolve(2)
+                    }
+                )
+            }
+        )
+    },
+    Promise(
+        function(resolve, reject) {
+            imp.wakeup(0.5,
+                function() {
+                    resolve(3)
+                }
+            )
+        }
+    )
 ];
 
 local p = Promise.serial(series);
 ```
 
-## Testing
-
-Repository contains [impUnit](https://github.com/electricimp/impUnit) tests and a configuration for [impTest](https://github.com/electricimp/impTest) tool.
-
-### TL;DR
-
-```bash
-cp .imptest .imptest-local
-nano .imptest-local # edit device/model
-imptest test -c .imptest-local
-```
-
-### Running Tests
+While in the following code p rejects in 1.5 seconds as all the Promises are instantiated at the same time:
```

Comment:
again this example doesn't reject, it resolves in 1.5 seconds with a value of 3.

Reply:
you are right. fixed.